src/rust/runtime/queue/mod.rs

```diff
@@ -54,28 +54,36 @@ impl IoQueueTable {
     /// Allocates a new entry in the target I/O queue descriptors table.
     pub fn alloc(&mut self, qtype: u32) -> QDesc {
         let index: usize = self.table.insert(qtype);
-        QDesc::from(index + Self::BASE_QD)
+
+        // Ensure that the allocation would yield to a safe conversion between usize to u32.
+        // Note: This imposes a limit on the number of open queue descriptors in u32::MAX.
+        assert!(
+            (index as u32) + Self::BASE_QD <= u32::MAX,
+            "I/O descriptors table overflow"
+        );
```

Comment:
Since `Slab` appears to allocate indexes sequentially, I think it's reasonable to just assert here as we're highly unlikely to ever exceed the max value.  Otherwise, we'd want `alloc()` to optionally return a failure code.

Reply:
Yes. Furthermore `Slab` does try to re-use empty entries.